### PR TITLE
Bump nodejs to v18

### PR DESCRIPTION
**Pull request by eseliger**
After various dependency updates in https://github.com/sourcegraph/sourcegraph/pull/45328 the webpack build finally doesn't break anymore and we can switch to node 18! Node 16 went out of support recently.

## Test plan

CI will tell, node is only used for building assets, not in prod.

## App preview:

- [Web](https://sg-web-es-bump-node-18.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.40 kb) | 0.08% (+12.15 kb) 🔺 | 0.10% (+11.75 kb) 🔺 | 0.68% (+5) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/es/bump-node-18/compare-report.html;tab=live_object) for a full comparison between the commits dbdaf78 and 19f664d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/dev/utils/constants.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@ import path from 'path'
 import { ROOT_PATH } from '@sourcegraph/build-config'
 
 export const STATIC_ASSETS_URL = '/.assets/'
-export const DEV_SERVER_LISTEN_ADDR = { host: 'localhost', port: 3080 } as const
-export const DEV_SERVER_PROXY_TARGET_ADDR = { host: 'localhost', port: 3081 } as const
+export const DEV_SERVER_LISTEN_ADDR = { host: '127.0.0.1', port: 3080 } as const
+export const DEV_SERVER_PROXY_TARGET_ADDR = { host: '127.0.0.1', port: 3081 } as const
```
Comment (eseliger) [inline client/web/dev/utils/constants.ts:7]: caddy talks to 127.0.0.1, not localhost and somehow node18 makes a distinction there

**Review by valerybugakov — APPROVED** [review]
Nice!

Should we also update the nodejs version here? cc @Strum355 
https://github.com/sourcegraph/sourcegraph/blob/5b81eb0c749860b011eac1e38134d87e7a036c7c/shell.nix#L51-L55

**Comment by burmudar** [comment]
Been running a few main-dry-runs since @eseliger pointed out the issue. It seems more often than not client tests seem to be timing out on CI. We've seen runs often being timed out by Buildkite after 1h. Weirdly, some runs were fast, < 5 min but so far in all my dry-runs it have timed out :| 

Trying to see if updating jest would help 🤷🏼

File: client/web/dev/utils/constants.ts (hunk 1 above)
Comment (Strum355) [reply client/web/dev/utils/constants.ts:7]: eh? Shouldnt they be the same thing?

Comment (eseliger) [reply client/web/dev/utils/constants.ts:7]: don't ask me I just flail until things work 😆

**Comment by burmudar** [comment]
> Been running a few main-dry-runs since @eseliger pointed out the issue. It seems more often than not client tests seem to be timing out on CI. We've seen runs often being timed out by Buildkite after 1h. Weirdly, some runs were fast, < 5 min but so far in all my dry-runs it have timed out :| 
> 
> 
> 
> Trying to see if updating jest would help 🤷🏼 

@eseliger found this build which is still on the old node that showed the same behavior... which means this seems more like Jest issue!

With that, I am approving while I look into Jest ... or stumble trying to

**Review by burmudar — APPROVED** [review]

**Pull request closed without merging**